src/py42/clients/securitydata.py

```diff
@@ -397,14 +402,32 @@ def _get_plans_in_node(destination, node):
     }
 
 
-def _parse_file_location_response(response):
-
-    for location in response[u"locations"]:
-        paths = []
+def _parse_file_location_response(locations):
+    devices = {}
+    for location in locations:
         file_name = location[u"fileName"]
+        file_path = u"{}{}".format(location[u"filePath"], file_name)
         device_id = location[u"deviceUid"]
-        paths.append(u"{}{}".format(location[u"filePath"], file_name))
-        yield device_id, paths
+        device_entry = devices.get(device_id)
+        if device_entry:
+            device_entry["paths"].append(file_path)
+        else:
+            devices[device_id] = {"deviceGuid": device_id, "paths": [file_path]}
+
+    return [devices[key] for key in devices]
+
+
+def _get_version_lookup_info(events):
+    for event in events:
+        device_guid = event[u"deviceUid"]
+        md5 = event[u"md5Checksum"]
+        sha256 = event[u"sha256Checksum"]
+        fileName = event[u"fileName"]
+        filePath = event[u"filePath"]
+
+        if device_guid and md5 and sha256 and fileName and filePath:
+            path = "{}{}".format(filePath, fileName)
+            return device_guid, md5, sha256, path
```

Comment:
here for all `events`, are we interested only in one version look up?? if there are multiple `events` and if all properties are defined for the first `event` rest would be ignored!!! is that ok?

Reply:
This is exactly what were doing before (we were just taking the fist event that was found). The only difference is that this makes sure that the event we found actually has all of these properties before attempting to use it (cloud events in particular will not have a device guid.

Reply:
Using only the first event should be fine because pds will give us other device locations to use for the stream if the one on the event does not work.